mcs/class/referencesource/System/net/System/URI.cs

```diff
@@ -1069,11 +1069,49 @@ private static void SetEscapedDotSlashSettings(UriSectionInternal uriSection, st
         private string GetLocalPath(){
             EnsureParseRemaining();
 
+            // https://bugzilla.xamarin.com/show_bug.cgi?id=58400
+            bool isFileUrlWithHost = (
+                (m_Info.Offset.Host != m_Info.Offset.Path) && 
+                // only file URLs
+                IsFile && 
+                // Paths that got reinterpreted as files would be treated
+                //  as if they have a host, like "/a/b" which maps to "file:///a/b/",
+                //  or "C:\a\b\" which maps to "file:///c:/a/b"
+                OriginalString.StartsWith("file://") &&
+                // file://localhost/x/y needs to produce /x/y
+                !IsLoopback
+            );
+
+            // Manually check for "hosts" that are just forward slashes.
+            if (isFileUrlWithHost) {
+                isFileUrlWithHost = false;
```

Comment:
Use string.IndexOf look up instead

Reply:
I don't think I can use ```IndexOf``` to do a ```!=``` like this